README.md

```diff
@@ -127,40 +129,100 @@ $sslcert = @("EOT")
 -----END CERTIFICATE-----
 |-EOT
 
-$dbpass = Sensitive(Deferred(conjur::secret, ['production/postgres/password', {
+$dbpass = Deferred(conjur::secret, ['production/postgres/password', {
   appliance_url => "https://my.conjur.org",
   account => "myaccount",
   authn_login => "host/myhost",
   authn_api_key => Sensitive("2z9mndg1950gcx1mcrs6w18bwnp028dqkmc34vj8gh2p500ny1qk8n"),
   ssl_certificate => $sslcert
-}]))
+}])
 ```
 
-#### Sensitive data type
+#### Example usage
 
-Note `conjur::secret` returns values wrapped in a `Sensitive` data type. In
-some contexts, such as string interpolation, it might cause surprising results
-(interpolating to `Sensitive [value redacted]`). This is intentional, as it
-makes it more difficult to accidentally mishandle secrets.
+```puppet
+node 'server-123' {
+  $db_password = Deferred(conjur::secret, ['inventory/db-password'])
+
+  file { '/tmp/creds.txt':
+    ensure => file,
+    mode => '0600',
+    content => $db_password,
+  }
+
+  file { '/tmp/creds.ini':
+    ensure => file,
+    mode => '0600',
+    content => Deferred('inline_epp', [
+      'password=<%= $db_password.unwrap %>',
+      { 'db_password' => $db_password }
+    ]),
+  }
+}
+```
 
-To use a secret as a string, you need to explicitly request it using the
-`unwrap` function; the result of the processing should be again wrapped in
-a `Sensitive` value.
+#### `Deferred` functions
 
-In particular, you should not pass unwrapped secrets as resource parameters
-if you can avoid it. Many resource types support `Sensitive` data type and
-handle it correctly. If a resource you're using does not, file a bug.
+This module _requires_ the use of Puppet v6+
+[`Deferred` functions](https://puppet.com/docs/puppet/6.17/deferring_functions.html)
+to ensure that the credential retrieval is fully handled on the agent. Since the
+resolution of variables is done on the agent _after_ the catalog is compiled, anything
+that requires the value of the variable during the compilation step (e.g. template
+compilation)
+[must also be wrapped in a `Deferred` invocation](https://puppet.com/docs/puppet/6.17/template_with_deferred_values.html).
+
+It is also important to note that you should make sure that you invoke the
+`conjur::secret` function using the proper `Deferred` syntax:
+
+```puppet
+# GOOD: Passing the parameters as an array
+Deferred(conjur::secret, ['production/postgres/password'])
+```
+
+```puppet
+# BAD: This will not work!
+Deferred(conjur::secret('production/postgres/password'))
+```
```

Comment:
should there also be a "bad" example where `Deferred` is missing? can we say something about it like `BAD: Will lead to XX kind of error`?

Reply:
Full coverage like that is part of the troubleshooting doc issue (#120). I think it's very well out of the scope for this exact PR.

Reply:
this is just one line that I think would help make this critical piece more clear - if we don't have the exact result handy, just saying that it's bad (and succinctly why, even if it's just "will error") would be valuable

Reply:
ok- will fix.